packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,59 @@
+import qs from 'query-string';
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scope?: string | string[];
+  resource?: string | string[];
+};
+
+enum ResponseType {
+  code = 'code',
+}
+
+const joinStrings = (stringOrStringArray?: string | string[]) =>
+  stringOrStringArray === undefined
+    ? ''
+    : Array.isArray(stringOrStringArray)
+    ? stringOrStringArray.join(' ')
+    : stringOrStringArray;
+
+const wrapInStringArray = (stringOrStringArray?: string | string[]): string[] =>
+  stringOrStringArray === undefined
+    ? []
+    : Array.isArray(stringOrStringArray)
+    ? stringOrStringArray
+    : [stringOrStringArray];
+
+export const generateSignInUri = ({
+  authorizationEndpoint,
+  clientId,
+  redirectUri,
+  codeChallenge,
+  state,
+  scope,
+  resource,
+}: SignInUriParameters) => {
+  const queryStringWithoutResource = qs.stringify({
+    client_id: clientId,
+    redirect_uri: [redirectUri],
+    code_challenge: codeChallenge,
+    code_challenge_method: 'S256',
+    state,
+    response_type: ResponseType.code,
+    prompt: 'consent',
+    scope: joinStrings(scope),
+  });
+
+  const queryStringWithOnlyResource = wrapInStringArray(resource).reduce(
```

Comment:
Maybe we can limit the `scope` type as `string[]`.
[Slack Discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1643439717664529)

Reply:
Simplified, PTAL